libraries/legacy_plugins/account_statistics/account_statistics_plugin.cpp

```diff
@@ -75,7 +75,7 @@ void account_statistics_plugin::plugin_initialize( const boost::program_options:
    {
       ilog( "account_stats plugin: plugin_initialize() begin" );
 
-      database().post_apply_operation.connect( [&]( const operation_notification& o ){ _my->on_operation( o ); } );
+      database().post_apply_operation_proxy( [&]( const operation_notification& o ){ _my->on_operation( o ); } );
```

Comment:
No, we should not be changing interfaces. It is ugly to require changing interfaces to support this. If we need a different interface, we should discuss changing that as another task.

Reply:
@mvandeberg The biggest problem is that most of member functions and variables defined in your code is public what is troublesome from interface maintenance point of view. Then I think we shall define public methods to hide internals. In this case, it was needed to be sure that every plugin has been correctly connected to the database signals. This allows us to centralize time-measurement when callbacks are executed by each plugin code, next to print time-distribution report gathered during reindex process.  
Also I think it is good time to change plugin interfaces because of AppBase release, which by design changed plugin structure and interfaces.

Reply:
I'll make an issue to clean up these interfaces. Each plugin has an abstract name() method. We should be using that instead of each plugin specifying manually. Perhaps in a macro.

Reply:
Maybe best way would be to pass into proxy methods (i.e. pre_apply_operation_proxy) plugin instance (and change `const std::string& plugin_name` argument to  `const abstract_plugin& plugin`. Then inside proxy method plugin name can be retrieved. This is also more flexible from future point of view, since it could be simpler to use other abstract_plugin data during registration without changing the interface.